File: java/NodoInstituto.java
package java;
import java.util.Objects;

public class NodoInstituto {
    private Instituto instituto;
    private NodoInstituto sig,ant;


    public NodoInstituto() {
        this.sig = null;
        this.ant = null;
    }

    public NodoInstituto(Instituto instituto) {
        this.instituto = instituto;
    }

    public Instituto getInstituto() {return this.instituto;}
    public void setInstituto(Instituto instituto) {this.instituto = instituto;}
    public NodoInstituto getSig() {return this.sig;}
    public void setSig(NodoInstituto sig) {this.sig = sig;}
    public NodoInstituto getAnt() {return this.ant;}
    public void setAnt(NodoInstituto ant) {this.ant = ant;}

    @Override
    public boolean equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof NodoInstituto)) {
            return false;
        }
        NodoInstituto nodoInstituto = (NodoInstituto) o;
        return Objects.equals(instituto, nodoInstituto.instituto) && Objects.equals(sig, nodoInstituto.sig) && Objects.equals(ant, nodoInstituto.ant);
    }

    @Override
    public int hashCode() {
        return Objects.hash(instituto, sig, ant);
    }

    
}

File: c#/Interfaz Grafica/Guia_Centros_Estudio_Superior.Core/Models/NodoInstituto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;


using System.IO;
namespace Guia_Centros_Estudio_Superior.Core.Models;
[Serializable]

public class NodoInstituto
{
    private Instituto instituto;
    private NodoInstituto sig,ant;

    public Instituto Instituto
    {
        get => instituto;
        set => instituto = value;
    }
    public NodoInstituto Sig
    {
        get => sig;
        set => sig = value;
    }
    public NodoInstituto Ant
    {
        get => ant;
        set => ant = value;
    }

    public NodoInstituto() {
        this.Sig = null;
        this.Ant = null;
    }

    public NodoInstituto(Instituto instituto) {
        this.Instituto = instituto;
    }

    public Instituto getInstituto() {return this.Instituto;}
    public void setInstituto(Instituto instituto) {this.Instituto = instituto;}
    public NodoInstituto getSig() {return this.Sig;}
    public void setSig(NodoInstituto sig) {this.Sig = sig;}
    public NodoInstituto getAnt() {return this.Ant;}
    public void setAnt(NodoInstituto ant) {this.Ant = ant;}



}
